pkg/controller/plan/kubevirt.go

```diff
@@ -925,12 +925,21 @@ func (r *KubeVirt) UpdateVmByConvertedConfig(vm *plan.VMStatus, pod *core.Pod, s
 		return
 	}
 
-	firmware, err := ovfparser.GetFirmwareFromConfig(string(vmConfigXML))
-	if err != nil {
-		return
-	}
+	switch r.Source.Provider.Type() {
+	case api.Ova:
+		firmware, erro := ovfparser.GetFirmwareFromConfig(string(vmConfigXML))
+		if erro != nil {
+			return
+		}
 
-	vm.Firmware = firmware
+		vm.Firmware = firmware
+	case api.VSphere:
+		os, erro := ovfparser.GetOperationSystemFromConfig(string(vmConfigXML))
```

Comment:
how about mapping the value of the operating system here instead of adding another step (`MapVirtV2vOs`)?

Reply:
indeed it looks like i forgot to call the builder here :)

Reply:
> but please look again how it's done now. i think you meant we should put the whole map here? IMO currently it's how it suppose to be done

no, I didn't mean to the location of the code but to the timing of mapping the value given from virt-v2v - currently we get the value of `os` and then call the builder to map it but this requires adding it to the interface of the Builder that affects all providers. I'd prefer to get the value already converted from the VM status

Reply:
making it in the interface does make sense to me (OVA later on?). do you mean to have it within the parser?

Reply:
yes, currently from the parser and later on from a utility function that returns the operating system from the vm configuration

Reply:
much better, thanks